Add snapcraft.yaml and wrapper

aidanharris opened the pull request:
### Fix or Enhancement?

This pull request is a WIP to package astro up as a [snap](https://snapcraft.io/) application. This will enable easy installation on Linux devices (desktops and / or servers) via a simple `snap install astro` (note: we'll have to register the astro name in the snap store first). We can also automatically build the snap package via the snapcraft [build service](https://build.snapcraft.io/).

To build the snap:

```sh
cd /path/to/astro
docker run -it --rm -v "$PWD:$PWD" -w $PWD/snap snapcore/snapcraft bash
apt update
snapcraft
```

This will produce a .snap file that can be installed on a snap enabled Linux device as follows:

`sudo snap install --devmode ./astro_0.1_amd64.snap`

To Do:

- [x] Register "astro" on the snap store
- [x] Run astro as a background daemon
- [ ] Enable and test confinement (sandboxing)
- [ ] Fix logging - Because snaps are read-only filesystems astro currently fails on startup as follows:

```sh
❯ /snap/astro/current/bin/astro
yarn start v0.24.5
$ cross-env NODE_ENV=production babel-node index.js | bunyan 
[2017-06-06T18:08:03.914Z]  INFO: Astro/7719 on harris: Astro is running on port 3000 (version=1.1.0)
[2017-06-06T18:08:03.915Z]  WARN: Astro/7719 on harris: Uncaught Exception. Shutting down. (version=1.1.0)
[2017-06-06T18:08:03.915Z]  WARN: Astro/7719 on harris: EROFS: read-only file system, open 'logs/error.log' (version=1.1.0)
Done in 0.87s.
```

I'll need to read-up on the correct way to log in a snap application to fix this (although logging to stdout seems to be working fine).

- [ ] All tests passed

### Environment
- OS: Linux
- Node version: 6.10.2

aidanharris committed:
Add snapcraft.yaml and wrapper

aidanharris committed:
Add astro-imports wrapper

aidanharris committed:
Fix PATH variable

aidanharris committed:
Patch logging

OmgImAlexis commented:
But sure if you're still using Gitter but I registered the name astro and it's now asking me to give it a config file. Not sure if there's a way to add you as a collaborator or use the one you've added here.

OmgImAlexis commented:
I'll be adding a cli wrapper around astro so if want to wait for that to be merged then you will be able to use `--log=console` and it'll change over. I also want to get babel to compile a single file ES5 file into a lib dir so we have a single file as the entry point and which means we won't need babel to run in production.

aidanharris committed:
Remove yarn and add daemons

Yarn is experimental (see: https://github.com/snapcore/snapcraft/releases/tag/2.29) and currently not working properly. Using npm although slower works fine including the background daemons that were broken with yarn because it wanted to write to the filesystem.
